lib/mergeset/table.go

```diff
@@ -1543,7 +1546,14 @@ func (tb *Table) CreateSnapshotAt(dstDir string) error {
 	if err != nil {
 		return fmt.Errorf("cannot read directory: %w", err)
 	}
-	for _, fi := range fis {
+
+	for i, fi := range fis {
+		if deadline > 0 && i%5 == 0 {
```

Comment:
are you sure that time is spent particularly in this loop?

Reply:
I've checked the code paths for snapshot creation and this is one of the places where we can do deadline check and which takes some time.
Another time-wise expensive operation will be `fsync` syscall, but it does not have an option to cancel it.